Content.Shared/Weapons/Ranged/Systems/SharedGunSystem.cs

```diff
@@ -304,6 +304,13 @@ private void AttemptShoot(EntityUid user, EntityUid gunUid, GunComponent gun)
 
         if (ev.Ammo.Count <= 0)
         {
+            // triggers effects on the gun when it's empty
+            if (gun.TriggerWhenEmpty)
+            {
+                TriggerGun(gunUid);
+                return;
+            }
```

Comment:
As above, just have a separate component subscribe instead of diluting gun code.

Reply:
I made it raise an event and subscribed TriggerSystem to it